training-service/src/app/pages/mail-form/mail-form.component.html

```diff
@@ -0,0 +1,21 @@
+<h2>E-mail Form Validation</h2>
+<form [formGroup]="mailForm">
+  <div class="form-group">
+    <input
+      type="email"
+      class="form-control"
+      placeholder="Enter e-mail"
+      formControlName="emailAddress"
+      [ngClass]="!mailForm.valid ? 'not-valid-input' : 'valid-input'"
```

Comment:
Are you sure you do not want to check touched prop also ?

Reply:
done.